### PR TITLE
Addition of some additional api (I am not sure these are clean since I am used to Qt)

### PR DESCRIPTION
**Pull request by bhattaraiashish**
Client::createGuildChannel got more parameters for channel creation
"bitrate" and "user_limit" is to be only sent when creating a voice channel otherwise the payload won't be accepted
parentId added for Discord::Channel
Discord::Embed fields are written to Json object
guildId retrieved from Message object
locale added to User object

### CODE REVIEW

File: src/core/Discord/HttpService.cpp [hunk 1]
```diff
@@ -6,6 +6,9 @@
 #include <QtCore/QBuffer>
 #include <QtCore/QJsonDocument>
 #include <QtNetwork/QNetworkReply>
+#include <QtNetwork/qhttpmultipart.h>
+#include <QtGui/QPixmap>
+#include <QtCore/qfile.h>
```
Comment (Gaztin) [inline src/core/Discord/HttpService.cpp]: Prefer using the "extension-less" includes! so QHttpMultiPart and QFile respectively here

**Review by Gaztin — CHANGES_REQUESTED** [review]
Looks good in general. I would like it if you could follow the coding style a bit more strictly. The thing that stands out most is the same-line braces!

File: src/core/Discord/HttpService.cpp (hunk 1 above)
Comment (bhattaraiashish) [reply src/core/Discord/HttpService.cpp]: ok

**Comment by bhattaraiashish** [comment]
I have made the said change.

**Comment by Gaztin** [comment]
Looks good!

**Pull request merged**